### PR TITLE
modal windon is ready to use

### PR DESCRIPTION
**Pull request by ElizarAitbek**

### CODE REVIEW

File: src/components/UI/ModalWindow.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+   modalTitle,
+   bodyContent,
+   headerContent,
+   footerContent,
+}) => {
+   const [open, setOpen] = useState(false)
+   const handleOpen = () => setOpen(true)
+   const handleClose = () => setOpen(false)
+   return (
+      <>
```
Comment (songhee24) [inline src/components/UI/ModalWindow.jsx:14]: modal must be without the button inside the modal component

File: src/components/UI/ModalWindow.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+         <Button onClick={handleOpen} variant="contained">
```
Comment (songhee24) [inline src/components/UI/ModalWindow.jsx]: Why we are using button inside modal component.?

File: src/components/UI/ModalWindow.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { Button, Grid, Modal, styled, Typography } from '@mui/material/'
+import { useState } from 'react'
+
+export const ModalWindow = ({
+   modalTitle,
+   bodyContent,
+   headerContent,
+   footerContent,
+}) => {
+   const [open, setOpen] = useState(false)
```
Comment (songhee24) [inline src/components/UI/ModalWindow.jsx]: that state will be created when someone will call your modal component
Also button which will open the modal

File: src/components/UI/ModalWindow.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   const handleClose = () => setOpen(false)
+   return (
+      <>
+         <Button onClick={handleOpen} variant="contained">
+            Contained
+         </Button>
+         <ModalMain open={open} onClose={handleClose}>
+            <ModalBlock>
+               <TitleBlock>{modalTitle}</TitleBlock>
+               <Grid>{headerContent}</Grid>
```
Comment (songhee24) [inline src/components/UI/ModalWindow.jsx]: Have you checked with other elements not only the text ?

**Review by songhee24 — CHANGES_REQUESTED** [review]

File: src/components/UI/ModalWindow.jsx (hunk 4 above)
Comment (ElizarAitbek) [reply src/components/UI/ModalWindow.jsx]: Проверял с текстом и внешними компонентами, каждая из них работает корректно.

**Pull request merged**